### PR TITLE
Step2: fixes #13781, fixes #13805 

### LINKED ISSUE #13781: gc:arc cannot fully support threadpool with FlowVar
# Example
```nim
import threadpool, os

proc thread(): string =
  os.sleep(1000)
  return "ok"

var fv = spawn thread()
sync()
echo ^fv
```

### Current Output
```
Error: cannot bind another '=destroy' to: FlowVarObj; previous declaration was constructed here implicitly: d:\nim\lib\pure\concurrency\threadpool.nim(211, 48)
```

### Expected Output
```
ok
```

$ nim -v
Nim Compiler Version 1.1.1 [Windows: amd64]
Compiled at 2020-03-27```

### LINKED ISSUE #13805: arc & threads: doesn't pass the reference forwarding test
The object is not destroyed:
```nim
import threadpool, os
type
  MyObj = object
    p: int

proc `=destroy`(x: var MyObj) = 
  if x.p != 0:
    echo "destroyed"

proc thread4(x: ref MyObj): ref MyObj =
  os.sleep(1000)
  result = x

proc ref_forwarding_test = 
  var x = new(MyObj)
  x[].p = 2
  var y = spawn thread4(x)
  sync()

ref_forwarding_test()
```

Destructor is not invoked.

### PR DESCRIPTION
**Pull request by cooldome**
Spawn for gc arc fixes

### CODE REVIEW

File: compiler/injectdestructors.nim [hunk 1]
```diff
@@ -791,6 +791,12 @@ proc p(n: PNode; c: var Con; mode: ProcessMode): PNode =
       if mode == normal and isRefConstr:
         result = ensureDestruction(result, c)
     of nkCallKinds:
+      let inSpawn = c.inSpawn
+      if n[0].kind == nkSym and n[0].sym.magic == mSpawn:
+        c.inSpawn.inc
+      elif c.inSpawn > 0:
```
Comment (Araq) [inline compiler/injectdestructors.nim:797]: Suspicious logic. Why not at the end of the ``of nkCallkinds`` handling code like

```nim

if n[0].kind == nkSym and n[0].sym.magic == mSpawn:
  c.inSpawn.dec
```

?

Comment (cooldome) [reply compiler/injectdestructors.nim:797]: The idea is that spawn can be written in the following way:
```nim
spawn myproc(call1(), (let x = call2(); call3(x, 0))
```
I want  `c.inSpawn` to be greater than zero only for `myproc` call but not for `call2` and `call3` in example above.

**Comment by cooldome** [comment]
I can't replicate CI test failures. I have tried 3 machines, 2 platforms and 3 compilers. All tests pass.
Don't know how to progress further

**Comment by Araq** [comment]
Well BSD says:

Category: arc
Name: tests/arc/tthread.nim C
Action: run
Result: reExitcodesDiffer
-------- Expected -------
exitcode: 0
--------- Given  --------
exitcode: 1

Output:

-------------------------

**Comment by cooldome** [comment]
Ready for review

File: lib/pure/concurrency/threadpool.nim [hunk 2]
```diff
@@ -234,43 +236,42 @@ proc awaitAndThen*[T](fv: FlowVar[T]; action: proc (x: T) {.closure.}) =
...
+    blockUntil(fv[])
     result = fv.blob
-  finished(fv)
+    finished(fv[])
+
+else:
+  proc `^`*[T](fv: FlowVar[T]): T =
+    ## Blocks until the value is available and then returns this value.
+    blockUntil(fv[])
+    when T is string or T is seq or T is ref:
```
Comment (metagn) [inline lib/pure/concurrency/threadpool.nim]: Why not get rid of the 2 different procs in each when/else branch and put `when defined(nimV2) and (T is string or T is seq or T is ref):` here?

Comment (cooldome) [reply lib/pure/concurrency/threadpool.nim]: done

**Pull request merged**